Client/Frontend/Home/FxHomeRecentlySavedCollectionCell.swift

```diff
@@ -0,0 +1,250 @@
+    static let bookmarkStackViewShadowRadius: CGFloat = 4
+    static let bookmarkStackViewShadowOffset: CGFloat = 2
+}
+
+/// A cell used in FxHomeScreen's Recently Saved section.
+class RecentlySavedBookmarkCell: UICollectionViewCell {
+    
+    // MARK: - Properties
+    
+    static let cellIdentifier = "recentlySavedBookmarkCell"
+    
+    // UI
+    var heroImage: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.contentMode = .scaleAspectFit
+        imageView.clipsToBounds = true
+        imageView.layer.masksToBounds = true
+        imageView.layer.cornerRadius = RecentlySavedBookmarkCellUX.generalCornerRadius
+        
+        return imageView
+    }()
+    var divider: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = UIColor.theme.homePanel.activityStreamCellDescription
+        
+        return view
+    }()
+    var bookmarkTitle: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontSizeToFitWidth = false
+        label.font = UIFont.systemFont(ofSize: RecentlySavedBookmarkCellUX.bookmarkTitleFontSize)
+        
+        return label
+    }()
+    var bookmarkDetails: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontSizeToFitWidth = false
+        label.font = UIFont.systemFont(ofSize: RecentlySavedBookmarkCellUX.bookmarkDetailsFontSize)
+        
+        return label
+    }()
+    
+    // MARK: - Inits
+    
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        
+        applyTheme()
+        setupObservers()
+        setupLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Helpers
+    
+    private func setupObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(handleNotifications), name: .DisplayThemeChanged, object: nil)
+    }
+    
+    private func setupLayout() {
+        contentView.layer.cornerRadius = RecentlySavedBookmarkCellUX.generalCornerRadius
+        contentView.layer.shadowRadius = RecentlySavedBookmarkCellUX.bookmarkStackViewShadowRadius
+        contentView.layer.shadowOffset = CGSize(width: 0, height: RecentlySavedBookmarkCellUX.bookmarkStackViewShadowOffset)
+        contentView.layer.shadowColor = UIColor.theme.homePanel.shortcutShadowColor
+        contentView.layer.shadowOpacity = UIColor.theme.homePanel.shortcutShadowOpacity
+        
+        contentView.addSubview(heroImage)
+        contentView.addSubview(divider)
+        contentView.addSubview(bookmarkTitle)
+        contentView.addSubview(bookmarkDetails)
+        
+        NSLayoutConstraint.activate([
```

Comment:
Don't know how other people feel about this, but big +1 from my POV for using native layout constraints.